lib/ansible/modules/cloud/amazon/elb_application_lb.py

```diff
@@ -452,173 +489,310 @@ def get_elb(connection, module):
+    if current_listener['Protocol'] == 'HTTPS' and new_listener['Protocol'] == 'HTTPS':
+        # Cert
+        if current_listener['SslPolicy'] != new_listener['SslPolicy']:
+            modified_listener['SslPolicy'] = new_listener['SslPolicy']
+        if current_listener['Certificates'][0]['CertificateArn'] != new_listener['Certificates'][0]['CertificateArn']:
+            modified_listener['Certificates'][0]['CertificateArn'] = new_listener['Certificates'][0]['CertificateArn']
+    elif current_listener['Protocol'] != 'HTTPS' and new_listener['Protocol'] == 'HTTPS':
+        modified_listener['SslPolicy'] = new_listener['SslPolicy']
+        modified_listener['Certificates'][0]['CertificateArn'] = new_listener['Certificates'][0]['CertificateArn']
+
+    # Default action
+    #   We wont worry about the Action Type because it is always 'forward'
+    if current_listener['DefaultActions'][0]['TargetGroupArn'] != new_listener['DefaultActions'][0]['TargetGroupArn']:
+        modified_listener['DefaultActions'] = []
+        modified_listener['DefaultActions'].append({})
+        modified_listener['DefaultActions'][0]['TargetGroupArn'] = new_listener['DefaultActions'][0]['TargetGroupArn']
+        modified_listener['DefaultActions'][0]['Type'] = 'forward'
+
+    if modified_listener:
+        return modified_listener
+    else:
+        return None
+
+
+def compare_condition(current_conditions, condition):
+    """
+
+    :param current_conditions:
+    :param condition:
+    :return:
+    """
+
+    condition_found = False
+
+    for current_condition in current_conditions:
+        if current_condition['Field'] == condition['Field'] and current_condition['Values'][0] == condition['Values'][0]:
+            condition_found = True
+            break
+
+    return condition_found
+
+
+def compare_rule(current_rule, new_rule):
+    """
+    Compare two rules.
+
+    :param current_rule:
+    :param new_rule:
+    :return:
+    """
+
+    modified_rule = {}
+
+    # Priority
+    if current_rule['Priority'] != new_rule['Priority']:
+        modified_rule['Priority'] = new_rule['Priority']
+
+    # Actions
+    #   We wont worry about the Action Type because it is always 'forward'
+    if current_rule['Actions'][0]['TargetGroupArn'] != new_rule['Actions'][0]['TargetGroupArn']:
+        modified_rule['Actions'] = []
+        modified_rule['Actions'].append({})
+        modified_rule['Actions'][0]['TargetGroupArn'] = new_rule['Actions'][0]['TargetGroupArn']
+        modified_rule['Actions'][0]['Type'] = 'forward'
+
+    # Conditions
+    modified_conditions = []
+    for condition in new_rule['Conditions']:
+        if not compare_condition(current_rule['Conditions'], condition):
+            modified_conditions.append(condition)
+
+    if modified_conditions:
+        modified_rule['Conditions'] = modified_conditions
+
+    return modified_rule
+
+
+def compare_listeners(connection, module, current_listeners, new_listeners, purge_listeners):
```

Comment:
I think I'd handle purge_listeners at the consumer end of this method. Just seems easier than passing it in and handling it here. But not a blocker